pkg/util/mount/mount.go

```diff
@@ -274,7 +276,7 @@ func IsNotMountPoint(mounter Interface, file string) (bool, error) {
 // The list equals:
 //   options - 'bind' + 'remount' (no duplicate)
 func isBind(options []string) (bool, []string) {
-	bindRemountOpts := []string{"remount"}
+	bindRemountOpts := []string{"bind", "remount"}
```

Comment:
This is what I had to do to make it work for the case that the volume is writable, but the subpath is read only.  

The original behavior of bind mount is: If bind mount is read only, and the underlying volume is writable, then this remount will also remount the original volume as read only. For this fix, if you also pass in the "bind" option to the remount, then it will make only the bind mount read only, and will not change the original volume mount.

We need to be sure this behavior change is ok for normal bind mounts that volume plugins do.

Reply:
just to confirm, previously we were still bind-mounting right? just not explicitly passing in this option, which as far as we know only changes behavior in this specific case?

Reply:
So this change actually "sort of" deviates from the article you linked, because it looks like in theory we should be able to get a read-only bindmount by doing (according to this article linked):
```
mount -o bind /vital_data /untrusted_container/vital_data
mount -o remount,ro /untrusted_container/vital_data
```

which we had before. However, that actually has the behavior of modifying the original mount point to also be read only, so what you've done is changed the command to the following to fix that:

```
mount -o bind /vital_data /untrusted_container/vital_data
mount -o bind,remount,ro /untrusted_container/vital_data
```

Let me know if that's a flawed understanding.

Reply:
Interesting thing I found:
https://unix.stackexchange.com/questions/198590/what-is-a-bind-mount

With Linux, the simple way:

mount --bind /some/where /mnt/readonly
mount -o remount,ro,bind /mnt/readonly
This leaves a short interval of time during which /mnt/readonly is read-write. If this is a security concern, first create the bind mount in a directory that only root can access, make it read-only, then move it to a public mount point. In the snippet below, note that it's important that /root/private (the directory above the mount point) is private; the original permissions on /root/private/mnt are irrelevant since they are hidden behind the mount point.

mkdir -p /root/private/mnt
chmod 700 /root/private
mount --bind /some/where /root/private/mnt
mount -o remount,ro,bind /root/private/mnt
mount --move /root/private/mnt /mnt/readonly

Reply:
Ah ah yes you're right.  I totally forgot to explain the issue I hit when it wasn't specifying the "bind" option in the remount.

Because subpath is a subdirectory of the volume, when you do the remount on the base volume as readonly, that remount will fail because of a "device is busy" error.  And I think that is because the subpath is a nested mount inside the base volume.

Reply:
Interesting, that's not the problem I expected, might be worth putting a comment above this line with a quick explanation or link to a description of why its necessary. Could be easily confused for a bug or misunderstood in the future.